server/core/src/catalog_utilities.cpp

```diff
@@ -85,32 +85,30 @@ namespace irods::experimental::catalog
             case entity_type::collection: {
                 nanodbc::statement stmt{_db_conn};
 
-                prepare(stmt, "select t.token_id from R_TOKN_MAIN t"
-                              " inner join R_OBJT_ACCESS a on t.token_id = a.access_type_id "
-                              "where"
-                              " a.user_id = (select user_id from R_USER_MAIN where user_name = ?) and"
-                              " a.object_id = ?");
+                prepare(stmt, "select distinct access_type_id from R_USER_GROUP ug "
+                              "inner join R_USER_MAIN u on ug.user_id = u.user_id "
+                              "inner join R_OBJT_ACCESS a on ug.group_user_id = a.user_id "
+                              "where u.user_name = ? and "
+                                    "a.object_id = ? and "
+                                    "a.access_type_id >= 1120"); // 1120 => "modify object"
```

Comment:
I'm not sure. I can see it going multiple ways. I'll state why I hard-coded the value. That might be the best place to start a discussion...

I hard-coded the value with the assumption that the permission model will not change for a very long time. If the permission model does change, my hope is that our tests would point to this query being an issue before we shipped bad code.

Using `access_type::modify_object` makes me think we're guarding against the integer values changing. I'm guessing that is the concern?

Hard-coding can lead to brittle code in the long run depending on what it is.

What do others think? @trel @SwooshyCueb

Reply:
My main reason for bringing it up was that I saw the comment off to the side and figured this could be an opportunity for "self-documenting code". I'm fine with the hard-coded value, too, given your reasoning.

Reply:
what's the upside of hard-coding?      seems we should use the nice things since we have them?

Reply:
The primary upside is fewer CPU instructions. This was purely going off of the idea that the permission model will likely never change. And by change, I mean the actual mapping between access names and integer values will not change.

Reply:
wouldn't it have already been compiled in?      oh... wait, this comes out of the database per agent?

Reply:
I don't understand what you mean by ... *this comes out of the database per agent?*

I'm starting to lean towards changing it to use the enumeration. The change won't impact runtime in any way and if we feel it will ease maintainability, then I'm all for it.

Reply:
doesn't the 1120 only live in the catalog?

Reply:
and yes, i think maintainability is good - and the speedup here, if any, is negligible.

Reply:
The 1120 is defined in the catalog and in the code. See `access_type` in catalog_utilities.hpp. There might be one other location.

I'll update it to use the enumeration.

Reply:
done.